BUG: Convert non-dates in xls date cells to number

kordek opened the pull request:
- [x ] closes #10001 
- [x ] tests added / passed
- [x ] passes `git diff upstream/master | flake8 --diff`
- [x ] whatsnew entry

closes GH10001

jreback commented:
pls add some tests

jreback commented on pandas/io/excel.py:
```diff
@@ -329,29 +329,38 @@ def _parse_cell(cell_contents, cell_typ):
             if cell_typ == XL_CELL_DATE:
-                if xlrd_0_9_3:
-                    # Use the newer xlrd datetime handling.
-                    cell_contents = xldate.xldate_as_datetime(cell_contents,
```
rather than a giant try: except: can you narrow down its placement

jreback commented on pandas/io/tests/test_excel.py:
```diff
@@ -481,6 +481,14 @@ def test_set_column_names_in_parameter(self):
+    def test_date_conversion_overflow(self):
+        # GH 10001 : pandas.ExcelFile ignore parse_dates=False
+        wrong_cell_value = 100000000000000000000
+        with warnings.catch_warnings(record=True) as w:
```
what are you catching with this warning?

kordek replied:
This is to test if the warning has been issued to the user.

jreback replied:
them use ``tm.assert_produces_warning(name_of_warning)``

jreback commented on pandas/io/tests/test_excel.py:
```diff
@@ -481,6 +481,14 @@ def test_set_column_names_in_parameter(self):
+        wrong_cell_value = 100000000000000000000
+        with warnings.catch_warnings(record=True) as w:
+            actual = self.get_exceldf('testdateoverflow')
+            parsed_value = actual.iloc[2, 0]
```
compare the actual value with ``assertEqual``

jreback replied:
use

```
result = ....
expected = ...
tm.assert....
```

kordek committed:
BUG: Convert non-dates in xls date cells to number

closes GH10001

If there is a date column in excel in which there are cells with some big
integers, that during parsing to date cause int/long overflow, issue a
warning and convert the value to int or float.

jreback commented on pandas/io/excel.py:
```diff
@@ -329,29 +329,38 @@ def _parse_cell(cell_contents, cell_typ):
+                        else:
+                            cell_contents = datetime(*dt)
+                except OverflowError:
+                    warn("\nCell marked as date caused C int overflow while "
```
actually this is not needed warning. Just leave the cell as it was. The result will end up being object dtype (if there are convertible dates in the column along with some non-convertible ones). Pls tests this as well.

kordek replied:
Ok will do this

kordek commented:
Prevoiusly failed tests on the lines that now appear as uncovered according to codecov, I'm not getting this.

jreback commented:
if you want to rebase I think I fixed codecov

kordek commented:
Thanks, seems ok now

jreback commented:
can u post a screen shot of what this excel file looks like

kordek commented:
![image](https://cloud.githubusercontent.com/assets/10603885/14943444/18f48172-0fda-11e6-947f-2d4da98caa47.png)

kordek commented:
![image](https://cloud.githubusercontent.com/assets/10603885/14943461/4eb6e75a-0fda-11e6-814a-29a223e7b572.png)

jreback commented on pandas/io/tests/test_excel.py:
```diff
@@ -481,6 +481,14 @@ def test_set_column_names_in_parameter(self):
+        # GH 10001 : pandas.ExcelFile ignore parse_dates=False
+        expected_value = 100000000000000000000
+        act_series = self.get_exceldf('testdateoverflow')['DateColWithBigInt']
+        act_value = act_series.iloc[2]
```
construct the expected frame and use assert_frame_equal

kordek commented:
Same thing with codecov

jreback commented:
you need to rebase on master

``git rebase -i orgin/master``

you are not picking up the change I made recently.

kordek commented:
check now pls

jreback commented on pandas/io/tests/test_excel.py:
```diff
@@ -481,6 +481,16 @@ def test_set_column_names_in_parameter(self):
+        # GH 10001 : pandas.ExcelFile ignore parse_dates=False
+        refdf = pd.DataFrame([[pd.Timestamp('2016-03-12'), 'Marc Johnson'],
+                              [pd.Timestamp('2016-03-16'), 'Jack Black'],
+                              [1e+20, 'Timothy Brown']],
```
I am think this would be converted as an integer, right? (or is it actually a float)

kordek replied:
In the current implementation it is a float (simply cell_value is returned). Or such way would be more desired ?: 
```python
val = int(cell_contents)
                if val == cell_contents:
                    cell_contents = val
```

jreback commented on doc/source/whatsnew/v0.18.1.txt:
```diff
@@ -656,3 +656,4 @@ Bug Fixes
 - Bug in ``to_numeric`` with ``Index`` returns ``np.ndarray``, rather than ``Index`` (:issue:`12777`)
 - Bug in ``to_numeric`` with datetime-like may raise ``TypeError`` (:issue:`12777`)
 - Bug in ``to_numeric`` with scalar raises ``ValueError`` (:issue:`12777`)
+- Bug where ``_parse_cell`` raised OverflowError if numeric value was placed in excel cell of type 'date'  (:issue:`10001`)
```
move to 0.18.2 (just about to release 0.18.1). make this a more meaningful message to the user.

jreback commented:
thanks @kordek
